### PR TITLE
Deprecate 'l1' and 'l2' in favor of 'absolute_error' and 'squared_error' (#82)

### LINKED ISSUE #82: Strings like 'l2' and 'l1' would be preferable as 'L2' and 'L1"
In skimming over the changes that black did I noticed the strings 'l1' and 'l2'. Normally for names lower case l is avoided as it can be mistaken for capital I and even for numeric 1 depending on font. I think it would be preferable to have these as uppercase L, since the docstring even talks about L2 loss where L is in capitals. This is in code like NetworkNeuralClassifier/Regressor among others. And where we test the value, if we choose to accept lower case for compatibility, then maybe do an upper() so it can be compared to 'L2' etc.

### PR DESCRIPTION
**Pull request by mattwright99**
### Summary

Closes #82 

### Details and comments

The `TrainableModel` class supports L1 and L2 loss functions and are chosen using the argument values `'l1'` and `'l2'`. Since this can be confusing, this PR deprecates these values by replacing occurances of `'l1'` with `'absolute_error'` and `'l2'` with `'squared_error'`. The PR also updates unit tests and documentation.

### CODE REVIEW

File: qiskit_machine_learning/algorithms/classifiers/neural_network_classifier.py [hunk 1]
```diff
@@ -56,15 +56,16 @@ def __init__(
                 as one sample and the loss function is applied to the whole vector. Otherwise, each
                 entry of the probability vector is considered as an individual sample and the loss
                 function is applied to the index and weighted with the corresponding probability.
-            loss: A target loss function to be used in training. Default is `l2`, i.e. L2 loss.
-                Can be given either as a string for 'l1', 'l2', 'cross_entropy',
-                'cross_entropy_sigmoid', or as a loss function implementing the Loss interface.
+            loss: A target loss function to be used in training. Default is `squared_error`, 
+                i.e. L2 loss. Can be given either as a string for 'asbolute_error' (i.e. L1 Loss),
```
Comment (woodsp-ibm) [inline qiskit_machine_learning/algorithms/classifiers/neural_network_classifier.py:60]:
```suggestion
                i.e. L2 loss. Can be given either as a string for 'absolute_error' (i.e. L1 Loss),
```

File: qiskit_machine_learning/algorithms/trainable_model.py [hunk 2]
```diff
@@ -51,9 +52,10 @@ def __init__(
                 as one sample and the loss function is applied to the whole vector. Otherwise, each
                 entry of the probability vector is considered as an individual sample and the loss
                 function is applied to the index and weighted with the corresponding probability.
-            loss: A target loss function to be used in training. Default is `l2`, i.e. L2 loss.
-                Can be given either as a string for 'l1', 'l2', 'cross_entropy',
-                'cross_entropy_sigmoid', or as a loss function implementing the Loss interface.
+            loss: A target loss function to be used in training. Default is `squared_error`, 
+                i.e. L2 loss. Can be given either as a string for 'asbolute_error' (i.e. L1 Loss),
```
Comment (woodsp-ibm) [inline qiskit_machine_learning/algorithms/trainable_model.py:56]:
```suggestion
                i.e. L2 loss. Can be given either as a string for 'absolute_error' (i.e. L1 Loss),
```

File: releasenotes/notes/l1-and-l2-loss-depreciation-5d7c37e7b459c4f6.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
+deprecations:
+  - |
+    The valid values passed to the loss argument of the ``TrainableModel`` constructor were partialy
```
Comment (woodsp-ibm) [inline releasenotes/notes/l1-and-l2-loss-depreciation-5d7c37e7b459c4f6.yaml:4]:
```suggestion
    The valid values passed to the loss argument of the ``TrainableModel`` constructor were partially
```

File: releasenotes/notes/l1-and-l2-loss-depreciation-5d7c37e7b459c4f6.yaml [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    deprecated (i.e. ``loss='l1'`` is replaced with ``loss='absolute_error'`` and ``loss='l2'`` is 
+    replaced with ``loss='squared_error'``). This affects instantiations of classes like the
```
Comment (woodsp-ibm) [inline releasenotes/notes/l1-and-l2-loss-depreciation-5d7c37e7b459c4f6.yaml:6]:
```suggestion
    replaced with ``loss='squared_error'``). This affects instantiation of classes like the
```

File: releasenotes/notes/l1-and-l2-loss-depreciation-5d7c37e7b459c4f6.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    ``NeuralNetworkClassifier``. This change was made to reduce confusion that stems from using
+    lowercase 'l' character which can be mistake for a numeric '1' or capital 'I'. You should
```
Comment (woodsp-ibm) [inline releasenotes/notes/l1-and-l2-loss-depreciation-5d7c37e7b459c4f6.yaml:8]:
```suggestion
    lowercase 'l' character which can be mistaken for a numeric '1' or capital 'I'. You should
```

File: releasenotes/notes/l1-and-l2-loss-depreciation-5d7c37e7b459c4f6.yaml [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+    update you model instantiations by replacing 'l1' with 'absolute_error' and 'l2' with
```
Comment (woodsp-ibm) [inline releasenotes/notes/l1-and-l2-loss-depreciation-5d7c37e7b459c4f6.yaml:9]:
```suggestion
    update your model instantiations by replacing 'l1' with 'absolute_error' and 'l2' with 
```

**Review by woodsp-ibm — COMMENTED** [review]
I will note we use `black` to format the code to avoid people having to manually fix style etc. Yo can run `make black`. It wants strings quoted with double quotes rather than single quotes. Hence though it looks like code is mostly ok, the build will be failing as black will complain about non-compliance. If you run make black locally it will change the files to conform and CI should then not fail in that regard.

**Comment by mattwright99** [comment]
@woodsp-ibm Yes I see now, thank you so much for the helful comments!

**Comment by woodsp-ibm** [comment]
CI still failing with some words it does not like `instantiation` and `instantiations`. Since the docstrings become our main API reference documentation we do the spell check to help keep up the quality. For words its built-in dictionary does not have, we have custom dict in the root here https://github.com/Qiskit/qiskit-machine-learning/blob/main/.pylintdict You will see they are in alphabetic order and lowercase. If you can add the above words to that file - yes, both the singular and plural forms are needed to be added there. That should get things through CI here. Thx

**Comment by manoelmarques** [comment]
I changed the code to use the deprecate_values decorator.

**Review by woodsp-ibm — APPROVED** [review]
LGTM. @mattwright99 Thanks for taking care of this!

**Review by manoelmarques — APPROVED** [review]

**Comment by mattwright99** [comment]
My pleasure! Thank you all so much all for helping me out with my first issue. Please feel free to let me know if there are any other beginner/intermediate tickets you want taken care of!

**Comment by adekusar-drl** [comment]
@mattwright99 Thanks a lot! If you fill comfortable you can take a look at #62 and #112. They are more interesting for sure, but in the same time more difficult for a beginner.

**Pull request merged**